x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -469,13 +470,26 @@ export const fillMissingCustomFields = ({
   const customFieldsKeys = new Set(customFields.map((customField) => customField.key));
   const missingCustomFields: CaseRequestCustomFields = [];
 
+  // only populate with the default value required custom fields missing from the request
   for (const confCustomField of customFieldsConfiguration) {
     if (!customFieldsKeys.has(confCustomField.key)) {
-      missingCustomFields.push({
-        key: confCustomField.key,
-        type: confCustomField.type,
-        value: null,
-      });
+      if (
+        confCustomField.required &&
+        confCustomField?.defaultValue !== null &&
+        confCustomField?.defaultValue !== undefined
+      ) {
+        missingCustomFields.push({
+          key: confCustomField.key,
+          type: confCustomField.type,
+          value: confCustomField.defaultValue,
+        } as CaseCustomField);
+      } else if (!confCustomField.required) {
+        missingCustomFields.push({
+          key: confCustomField.key,
+          type: confCustomField.type,
+          value: null,
+        } as CaseCustomField);
+      } // else, missing required custom fields without default are not touched
```

Comment:
Just curious about this 🙂 . Before, we were 'filling" (impossible due to validation) required fields with `null`. What is your thought process here?

Reply:
Good question, I thought a bit about this when implementing!

This function has to responsibilities:
1. fill missing optional custom fields
2. fill missing required custom fields with default value

Not just because of validation but logic-wise I thought it wouldn't make sense to do any extra thing here 🤷

Reply:
Thanks for the explanation. Make sense to me.